fsm-calculator/src/main/java/org/egov/fsm/calculator/services/FSMService.java

```diff
@@ -0,0 +1,65 @@
+package org.egov.fsm.calculator.services;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+
+import org.egov.fsm.calculator.config.CalculatorConfig;
+import org.egov.fsm.calculator.repository.ServiceRequestRepository;
+import org.egov.fsm.calculator.utils.CalculatorConstants;
+import org.egov.fsm.calculator.web.models.FSM;
+import org.egov.fsm.calculator.web.models.FSMResponse;
+import org.egov.fsm.calculator.web.models.RequestInfoWrapper;
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.tracer.model.CustomException;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+
+@Service
+public class FSMService {
+
+	@Autowired
+	private ServiceRequestRepository serviceRequestRepository;
+
+	@Autowired
+	private ObjectMapper mapper;
+
+	@Autowired
+	private CalculatorConfig config;
+
+	public FSM getFsmApplication(RequestInfo requestInfo, String tenantId, String applicationNo) {
+		StringBuilder url = getBPASearchURL();
+		url.append("tenantId=");
+		url.append(tenantId);
+		
+		url.append("&");
+		url.append("applicationNo=");
+		LinkedHashMap responseMap = null;
+		responseMap = (LinkedHashMap) serviceRequestRepository.fetchResult(url, new RequestInfoWrapper(requestInfo));
+
+		FSMResponse fsmResponse = null;
+
+		try {
+			fsmResponse = mapper.convertValue(responseMap, FSMResponse.class);
+		} catch (IllegalArgumentException e) {
+			throw new CustomException(CalculatorConstants.PARSING_ERROR, "Error while parsing response of FSM Application Search");
+		}
+
+		return fsmResponse.getFsm().get(0);
```

Comment:
handle the case in which search response is empty. Throw proper error like application not found

Reply:
Taken care